discord/poll.py

```diff
@@ -0,0 +1,624 @@
+
+            This can only be called when the parent poll has an attached message.
+
+        Examples
+        --------
+
+        Usage ::
+
+            async for voter in poll_answer.voters():
+                print(f'{voter} has voted for {poll_answer}!')
+
+        Flattening into a list: ::
+
+            voters = [voter async for voter in poll_answer.voters()]
+            # voters is now a list of User
+
+        Parameters
+        ----------
+        limit: Optional[:class:`int`]
+            The maximum number of results to return.
+            If not provided, returns all the users who
+            voted to this poll answer.
+        after: Optional[:class:`abc.Snowflake`]
+            For pagination, voters are sorted by member.
+
+        Raises
+        ------
+        HTTPException
+            Retrieving the users failed.
+
+        Yields
+        ------
+        Union[:class:`User`, :class:`Member`]
+            The member (if retrievable) or the user that has voted
+            to this poll answer. The case where it can be a :class:`Member`
+            is in a guild message context. Sometimes it can be a :class:`User`
+            if the member has left the guild.
+        """
+
+        if not self._message or not self._state:  # Make type checker happy
+            raise RuntimeError('You cannot fetch users in a non-message-attached poll')
+
+        if limit is None:
+            if not self._message.poll:
+                limit = 100
+            else:
+                limit = self.vote_count or 100
+
+        while limit > 0:
+            retrieve = min(limit, 100)
+
+            message = self._message
+            guild = self._message.guild
+            state = self._state
+            after_id = after.id if after else None
+
+            data = await state.http.get_poll_answer_voters(
+                message.channel.id, message.id, self.id, after=after_id, limit=retrieve
+            )
+            users = data['users']
+
+            if len(users) == 0:
+                # No more voters to fetch, terminate process
+                break
+
+            limit -= len(users)
+            after = Object(id=int(users[-1]['id']))
+
+            if not guild or isinstance(guild, Object):
+                for raw_user in reversed(users):
+                    yield User(state=self._state, data=raw_user)
+                continue
+
+            for raw_member in reversed(users):
+                member_id = int(raw_member['id'])
+                member = guild.get_member(member_id)
+
+                yield member or User(state=self._state, data=raw_member)
+
+
+class Poll:
+    """Represents a message's Poll.
+
+    .. container:: operations
+
+        .. describe:: str(x)
+
+            Returns the Poll's question
+
+        .. describe:: len(x)
+
+            Returns the Poll's answer amount.
+
+    .. versionadded:: 2.4
+
+    Parameters
+    ----------
+    question: Union[:class:`PollMedia`, :class:`str`]
```

Comment:
If the parameter type and the attribute types are different then you need a dedicated attribute section as well.